include/AST/Expr/LiteralExpr.hpp

```diff
@@ -0,0 +1,26 @@
+#ifndef GLU_AST_EXPR_LITERAL_EXPR_KIND_HPP
+#define GLU_AST_EXPR_LITERAL_EXPR_KIND_HPP
+
+#include "ASTNode.hpp"
+#include "Types.hpp"
+
+namespace glu::ast {
+
+class LiteralExpr : public ExprBase {
+    glu::types::TypeBase *_type;
+
```

Comment:
a literal has a value: int literals (APInt), float literals (APFloat), stringref for string literals, bool for booleans...